docs/developer/decision-records/2024-07-03-additional-catalogrequest-param/README.md

```diff
@@ -0,0 +1,44 @@
+# Adding an additional parameter to the `CatalogRequest` (Management API)
+
+## Decision
+
+The `POST /catalog/request` endpoint to request a catalog will receive a new optional field in the request body with which the requesting participant (=consumer) can insert additional scopes into the DCP interaction.
+
+## Rationale
+
+In typical DCP interactions, the consumer derives the required scopes from the current request (e.g. policies, or existing agreements). This does not work for a Catalog request, because there is no request context.
+
+Further, in many dataspaces there are "default scopes", i.e. scopes that must be present on every DSP interaction, but these are static - they can't be changed at runtime.
+
+So if a provider offers assets, that are only available is a special (non-default) credential is presented, the consumer must be able to attach the respective scope strings to the access token.
+
+_NB: the information \_which_ scopes must be added, has to be conveyed out-of-band.\_
+
+In other words, a consumer may know that a provider has certain assets available, but they have an access policy constraint on them, which makes them "invisible" unless a certain credential is presented.
+
+## Approach
+
+The `CatalogRequest` will receive a new field `parameters`, which is an extensible map. This map may contain an entry `additionalScopes` which is a list of scope strings (as per [DCP Specification, Section 3.1](https://github.com/eclipse-tractusx/identity-trust/blob/main/specifications/verifiable.presentation.protocol.md#31-access-scopes)):
+
+```json
+{
+  "@type": "CatalogRequest",
+  "counterPartyAddress": "http://provider-address.com",
+  "counterPartyId": "providerId",
+  "protocol": "dataspace-protocol-http",
+  "querySpec": {
+    //...
+  },
+  "parameters": {
```

Comment:
is this `parameters` really needed? `additionalScopes` could stay at the root level, as `CatalogRequest` itself can already be considered "extensible" (`querySpec` is not mandatory so in fact it represent an object extension)

Reply:
it's not strictly needed, but it would fit the paradigm we often use. Also, if we have this container object, we could easily add other features in the future without changing the shape of the body (i.e. the `CatalogRequest` object)

Reply:
it would be an extensible map, so we wouldn't need to change it. only whoever consumes the `CatalogRequest` objects needs to know to call `get(newParamterKey)`

Reply:
i think we are getting too technical for this scope, but nothing prevents extensible properties to be at the root level of an object (e.g. `Policy.extensibleProperties`, I know, they are not really used, but they are deserialized from the root level of a policy)

Reply:
if we use a `parameters` object, we only need to change the transformers and the API once, and the problem is solved for all future parameters. 
If we decide to remove the `additionalScopes` later, we could simply leave the outer wrapper object in place and not change the API layer, so no change. My objective was future-proofing this as much as possible.

Reply:
this should not be the case, if in the future we'll remove the `additionalScopes` in fact we'll need to change the API documentation, and that's should be transparent and described by a new version, because that's a breaking change, otherwise would be a sort of swept under the carpet.

"open" parameters are good when there's no knowledge in the domain about structure and content (see asset properties), but since we know how these `additionalScopes` are shaped and we know how to deserialize them, I would describe that clearly in our domain object (`CatalogRequest`).

Reply:
ok, let's agree to disagree, and not drag this on for too long. We're ultimately debating the dots on the "i"s. 
I'm just gonna put it directly into the root object.